### DISCUSSION TITLE
Exporting component props syntax error

### DISCUSSION THREAD

**Discussion by specialdoom** [solidjs/solid#798] [Q&A]
Hi. Keep getting this syntax error regarding the named export of a component props. 

Let's say I have a `Button.jsx` file where I have the Button component implementation and and `index.tsx` file where I import and export the `Button` component and `ButtonProps` interface.

When I'm using the `Button` component inside `App` component, I'm getting the following syntax error:

_Uncaught_ SyntaxError: The requested module '/src/button/Button.tsx' does not provide an export named _'ButtonProps'_

Anybody know why? Is there something wrong with this approach?

Stackblitz: https://stackblitz.com/edit/solid-vite-1a4ae6

**Comment by ryansolid** [comment]
Hmm.. I see what the problem is but I'm unclear why that setup has that issue. There is a setting that tells TypeScript to only transform type import/exports. We set that sometimes to make directives not disappear but it means that types can't be import/exported without using the type keyword. I don't see the setting and to my knowledge it's not the default but none the less changing index.jsx to this seems to fix it:

```ts
export { Button } from './Button';
export type { ButtonProps } from './Button';
```
